src/rpc_client/src/meta_client.rs

```diff
@@ -883,8 +924,203 @@ impl GrpcMetaClient {
+        let init_election_client = core_ref.lock().await.election_client.clone();
+        members.insert(current_leader.clone(), init_election_client);
+
+        struct ElectionMemberManagement {
+            core_ref: Arc<Mutex<GrpcMetaClientCore>>,
+            members: HashMap<String, LeaderServiceClient<Channel>>,
+            current_leader: String,
+        }
+
+        impl ElectionMemberManagement {
+            fn host_address_to_url(addr: &HostAddress) -> String {
+                format!("http://{}:{}", addr.host, addr.port)
+            }
+
+            async fn recreate_core(&self, channel: Channel) {
+                let mut core = self.core_ref.lock().await;
+                *core = GrpcMetaClientCore::new(channel);
+            }
+
+            async fn tick(&mut self) -> Result<()> {
+                self.refresh_members()
+                    .await
+                    .context("error happened when refresh election members")?;
+                self.refresh_leader()
+                    .await
+                    .context("error happened when refresh election leader")?;
+                Ok(())
+            }
+
+            async fn refresh_members(&mut self) -> Result<()> {
+                let mut members = None;
+                for (addr, client) in &self.members {
+                    let members_resp = match client.to_owned().members(MembersRequest {}).await {
+                        Ok(resp) => resp.into_inner(),
+                        Err(e) => {
+                            tracing::warn!("failed to fetch members from {}, {}", addr, e);
+                            continue;
+                        }
+                    };
+
+                    let member_addrs: HashSet<_> = members_resp
+                        .members
+                        .iter()
+                        .cloned()
+                        .flat_map(|member| member.member_addr)
+                        .map(|addr| Self::host_address_to_url(&addr))
+                        .collect();
+
+                    members = Some(member_addrs);
+                    break;
+                }
+
+                let member_addrs = match members {
+                    None => return Err(RpcError::Internal(anyhow!("could not refresh members"))),
+                    Some(members) => members,
+                };
+
+                let mut new_members = HashMap::new();
+
+                for addr in member_addrs {
+                    if let Some(client) = self.members.remove(&addr) {
+                        new_members.insert(addr, client);
+                    } else {
+                        let channel = GrpcMetaClient::build_rpc_channel(addr.as_str()).await?;
+                        new_members.insert(addr, LeaderServiceClient::new(channel));
+                    }
+                }
+
+                self.members = new_members;
+
+                Ok(())
+            }
+
+            async fn refresh_leader(&mut self) -> Result<()> {
```

Comment:
We can simply the member RPC service and introduce role for all members, so that we can refresh member and leader info in a single RPC call. Not emergency we can do it in anther PR.